NodeDependencyLookup/Editor/Caches/AssetDependencyCache/Resolvers/ObjectSerializedDependencyResolver.cs

```diff
@@ -9,54 +13,157 @@ public class AssetToAssetObjectDependency
+		private HashSet<string> ExcludedProperties = new HashSet<string>(new[] {"m_CorrespondingSourceObject"});
+
+		// Don't include any dependencies to UnityEngine internal scripts
+		private HashSet<string> ExcludedDependencies =
+			new HashSet<string>(new[] {"UnityEngine.UI.dll", "UnityEngine.dll"});
+
+		private Dictionary<string, string> cachedAssetAsDependencyData = new Dictionary<string, string>();
+
+		public void TraversePrefab(ResolverDependencySearchContext searchContext, Object obj, Stack<PathSegment> stack)
+		{
+			AddPrefabAsDependency(searchContext, obj, stack);
+		}
+
+		public void TraversePrefabVariant(ResolverDependencySearchContext searchContext, Object obj,
+			Stack<PathSegment> stack)
 		{
-			List<Dependency> subSystemDependencies = new List<Dependency>();
-			HashSet<string> foundDependenciesHashSet = new HashSet<string>();
+			stack.Push(new PathSegment("Variant Of", PathSegmentType.Component));
+			AddPrefabAsDependency(searchContext, obj, stack);
+			stack.Pop();
+		}
+
+		private void AddPrefabAsDependency(ResolverDependencySearchContext searchContext, Object obj,
+			Stack<PathSegment> stack)
+		{
+			var correspondingObjectFromSource = PrefabUtility.GetCorrespondingObjectFromSource(obj);
+			var assetId = NodeDependencyLookupUtility.GetAssetIdForAsset(correspondingObjectFromSource);
+			var assetPath = AssetDatabase.GetAssetPath(correspondingObjectFromSource);
+			var guid = AssetDatabase.AssetPathToGUID(assetPath);
 
-			GetDependenciesForIdFromSerializedPropertyTraverser(assetId, subSystemDependencies);
-			
-			foreach (Dependency dependency in subSystemDependencies)
+			if (guid != NodeDependencyLookupUtility.GetGuidFromAssetId(searchContext.AssetId))
 			{
-				foundDependenciesHashSet.Add(dependency.Id);
-				dependencies.Add(dependency);
+				searchContext.AddDependency(this,
+					new Dependency(assetId, AssetToAssetObjectDependency.Name, AssetNodeType.Name, stack.ToArray()));
 			}
 		}
 
-		public void GetDependenciesForIdFromSerializedPropertyTraverser(string assetId, List<Dependency> dependencies)
+		private string GetAssetPathForAsset(string sourceAssetId, Object value)
 		{
-			if (TraverserSubSystem.Dependencies.ContainsKey(assetId))
+			if (value == null)
+				return null;
+
+			var assetPath = AssetDatabase.GetAssetPath(value);
+
+			if (string.IsNullOrEmpty(assetPath) || ExcludedDependencies.Contains(Path.GetFileName(assetPath)))
 			{
-				foreach (var foundDependency in TraverserSubSystem.Dependencies[assetId])
-				{
-					string dependency = foundDependency.Id;
+				return null;
+			}
 
-					// Avoid adding node itself as dependency
-					if (dependency != assetId)
-					{
-						dependencies.Add(foundDependency);
-					}
-				}
+			var assetId = NodeDependencyLookupUtility.GetAssetIdForAsset(value);
+			var guid = NodeDependencyLookupUtility.GetGuidFromAssetId(assetId);
+
+			var isUnityAsset = guid.StartsWith("0000000");
```

Comment:
Is this your own convention or from Unity?

Reply:
It is my own. I know its not perfect but all guids for Unity assets start like this.